op-e2e/actions/upgrades/interop_fork_test.go

```diff
@@ -0,0 +1,144 @@
+package upgrades
+
+import (
+	"math/big"
+	"testing"
+
+	"github.com/ethereum-optimism/optimism/op-e2e/actions/helpers"
+	"github.com/ethereum-optimism/optimism/op-e2e/actions/interop/dsl"
+	"github.com/ethereum-optimism/optimism/op-e2e/actions/upgrades/utils"
+	"github.com/ethereum-optimism/optimism/op-node/rollup/derive"
+	"github.com/ethereum-optimism/optimism/op-node/rollup/event"
+	"github.com/ethereum-optimism/optimism/op-service/predeploys"
+	"github.com/ethereum/go-ethereum/common/hexutil"
+	"github.com/ethereum/go-ethereum/core/types"
+	"github.com/stretchr/testify/require"
+)
+
+func TestInteropUpgrade(gt *testing.T) {
+	t := helpers.NewDefaultTesting(gt)
+
+	// Run Isthmus until the interop upgrade
+	is := dsl.SetupInterop(t, dsl.SetInteropOffsetForAllL2s(uint64(15)))
+	actors := is.CreateActors()
+	actors.PrepareAndVerifyInitialState(t)
+
+	////////////////////////////
+	// Pre-upgrade Block Production
+	////////////////////////////
+
+	// Start op-nodes
+	actors.ChainA.Sequencer.ActL2PipelineFull(t)
+	actors.ChainB.Sequencer.ActL2PipelineFull(t)
+
+	rollupConfigA := is.Out.L2s[actors.ChainA.ChainID.String()].RollupCfg
+	rollupConfigB := is.Out.L2s[actors.ChainB.ChainID.String()].RollupCfg
+
+	// Verify Interop is not active at genesis yet
+	l2Head := actors.ChainA.Sequencer.L2Unsafe()
+	require.NotZero(t, l2Head.Hash)
+	require.True(t, rollupConfigA.IsIsthmus(l2Head.Time), "Isthmus should be active at genesis in chain A")
+	require.False(t, rollupConfigA.IsInterop(l2Head.Time), "Interop should not be active at genesis in chain A")
+
+	l2Head = actors.ChainB.Sequencer.L2Unsafe()
+	require.NotZero(t, l2Head.Hash)
+	require.True(t, rollupConfigB.IsIsthmus(l2Head.Time), "Isthmus should be active at genesis in chain B")
+	require.False(t, rollupConfigB.IsInterop(l2Head.Time), "Interop should not be active at genesis in chain B")
+
+	// Submit all and mine for both chains
```

Comment:
Might be missing something but I'm not seeing any batch submission or l1 confirmation on the next few lines? Do you mean to say "progress the L2 chain" or something?

Reply:
I had opted to do batch submission all at once below, but can do batch submission here as well if you'd prefer.

Reply:
Will fix in follow up PR, which was preemted by the need to merge this before the rc betanet upgrade rehearsal.

Reply:
I think it makes sense to batch&mine later, as it is currently the case. So let's just remove this stale comment.

Reply:
TODO in follow-up